tests/e2e/tkgs_ha.go

```diff
@@ -488,4 +494,1214 @@ var _ = ginkgo.Describe("[csi-tkgs-ha] Tkgs-HA-SanityTests",
+
+			pv := persistentvolumes[0]
+			volHandle := getVolumeIDFromSupervisorCluster(pv.Spec.CSI.VolumeHandle)
+			gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+			svcPVCName := pv.Spec.CSI.VolumeHandle
+			svcPVC := getPVCFromSupervisorCluster(svcPVCName)
+
+			defer func() {
+				err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				ginkgo.By("Verify PVs, volumes are deleted from CNS")
+				err = fpv.WaitForPersistentVolumeDeleted(client, pv.Name, framework.Poll,
+					framework.PodDeleteTimeout)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				volumeID := pv.Spec.CSI.VolumeHandle
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(volumeID)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+					fmt.Sprintf("Volume: %s should not be present in the CNS after it is deleted from "+
+						"kubernetes", volumeID))
+
+			}()
+
+			ginkgo.By("Verify SV storageclass points to GC storageclass")
+			gomega.Expect(svcPVC.Spec.StorageClassName == &storageclass.Name).To(
+				gomega.BeTrue(), "SV storageclass does not match with gc storageclass")
+			framework.Logf("GC PVC's storageclass matches SVC PVC's storageclass")
+
+			ginkgo.By("Create a pod and wait for it to come to Running state")
+			pod, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, "")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			defer func() {
+				ginkgo.By("Delete pod")
+				err = fpod.DeletePodWithWait(client, pod)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}()
+
+			ginkgo.By("Verify annotations on SVC PV and required node affinity details on SVC PV and GC PV")
+			ginkgo.By("Verify pod gets scheduled on appropriate nodes preset in the availability zone")
+			verifyAnnotationsAndNodeAffinity(allowedTopologyHAMap, categories, pod, nodeList, svcPVC, pv, svcPVCName)
+
+			verifyOfflineVolumeExpansionOnGc(client, pvclaim, svcPVCName, namespace, volHandle, pod, pv, f)
+
+		})
+
+		/*
+			Static volume provisioning using zonal storage
+			1. Create a zonal storage policy, on the datastore that is shared only to specific cluster
+			2. Use the Zonal storage class and Immediate binding mode
+			3. Create a FCD in SVC namespace
+			4. Create CNS register volume CRD to statically import PVC on SVC namespace
+			5. Verify the statically created svc-pv and svc-pvc
+			6. In Gc, Create Static gc-pv pointing the volume handle  of svc-pvc  and reclaim policy as retain
+			7. Verify the node affinity of gc1-pv and svc-pv , It should have node affinity of all the zones
+			8. Describe SVC-PV , and GC-PV  and verify node affinity, make sure appropriate node affinity gets added
+			9. Delete gc1-pv and gc1-pvc
+			10.Wait  for the PODs to reach running state - make sure Pod scheduled on
+			   appropriate nodes preset in the availability zone
+			11.Delete PVC,POD,SC
+		*/
+		ginkgo.It("Static volume provisioning using zonal storage", func() {
```

Comment:
please rebase, this test is  already merged

Reply:
Done